web_src/js/features/repo-settings.js

```diff
@@ -81,4 +82,28 @@ export function initRepoSettingBranches() {
     const $target = $($(this).attr('data-target'));
     if (this.checked) $target.addClass('disabled'); // only disable, do not auto enable
   });
+
+  // show the `Matched` mark for the status checks that match the pattern
+  const markMatchedStatusChecks = () => {
+    const patterns = document.getElementById('status_check_contexts')?.value.split(/[\r\n]+/);
+    const validPatterns = patterns.map((item) => item.trim()).filter((item) => item.length > 0);
+    const marks = document.getElementsByClassName('status-check-matched-mark');
+    for (const el of marks) {
+      let matched = false;
+      const statusCheck = el.getAttribute('data-status-check');
+      for (const pattern of validPatterns) {
+        if (minimatch(statusCheck, pattern)) {
+          matched = true;
+          break;
+        }
+      }
+      if (matched) {
+        el.classList.remove('gt-hidden');
+      } else {
+        el.classList.add('gt-hidden');
+      }
+    }
+  };
+  markMatchedStatusChecks();
+  document.getElementById('status_check_contexts')?.addEventListener('input', markMatchedStatusChecks);
```

Comment:
Wrap the callback with a `onInputDebounce` from `utils/dom.js`.

Reply:
done in 2f4c1a3